From: Sergei Shtylyov <sergei.shtylyov@cogentembedded.com>
Date: Wed, 30 Oct 2019 21:53:03 +0300
Subject: mtd: spi-nor: fix silent truncation in spi_nor_read_raw()
References: bsc#1175416

Git-commit: 3d63ee5deb466fd66ed6ffb164a87ce36425cf36
Patch-mainline: v5.5-rc1

spi_nor_read_raw() assigns the result of 'ssize_t spi_nor_read_data()'
to the 'int ret' variable, while 'ssize_t' is a 64-bit type and *int*
is a 32-bit type on the 64-bit machines. This silent truncation isn't
really valid, so fix up the variable's type.

Fixes: f384b352cbf0 ("mtd: spi-nor: parse Serial Flash Discoverable Parameters (SFDP) tables")
Signed-off-by: Sergei Shtylyov <sergei.shtylyov@cogentembedded.com>
Signed-off-by: Tudor Ambarus <tudor.ambarus@microchip.com>
Signed-off-by: Mian Yousaf Kaukab <ykaukab@suse.de>
---
 drivers/mtd/spi-nor/spi-nor.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/mtd/spi-nor/spi-nor.c
+++ b/drivers/mtd/spi-nor/spi-nor.c
@@ -1794,7 +1794,7 @@ static int spi_nor_read_sfdp(struct spi_
 			     size_t len, void *buf)
 {
 	u8 addr_width, read_opcode, read_dummy;
-	int ret;
+	ssize_t ret;
 
 	read_opcode = nor->read_opcode;
 	addr_width = nor->addr_width;

